app/src/main/java/com/endava/flights/MainFlightViewModel.kt

```diff
@@ -0,0 +1,22 @@
+package com.endava.flights
+
+import androidx.lifecycle.ViewModel
+import com.endava.flights.datasource.FlightImpl
+import com.endava.flights.model.GetFlightsByServicesUseCase
+import com.endava.flights.model.Service
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class MainFlightViewModel : ViewModel() {
+    private lateinit var getCheapestFlightsUseCase : GetFlightsByServicesUseCase
```

Comment:
Here we talked about the SOLID last principle. Try to figure out how to change this to inverse the responsibility. Here the object ViewModel is creating himself.